pilot/pkg/config/mcp/coredatamodel_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package coredatamodel_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	google_protobuf "github.com/gogo/protobuf/types"
+	"github.com/onsi/gomega"
+	mcp "istio.io/api/mcp/v1alpha1"
+	networking "istio.io/api/networking/v1alpha3"
+	mcpclient "istio.io/istio/galley/pkg/mcp/client"
+	coredatamodel "istio.io/istio/pilot/pkg/config/mcp"
+	"istio.io/istio/pilot/pkg/config/mcp/fakes"
+	"istio.io/istio/pilot/pkg/config/memory"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+func TestCoreDataModelUpdate(t *testing.T) {
```

Comment:
Please focus on integration tests - there is very little logic in the code that needs this kind of testing, and 
the problems we've had with adapters were integration related. 

It is very likely this code will be changed many times ( data structs, event model will change quite a bit), and 
people working on that will have to deal with complicated tests tied to the implementation details.

Reply:
If you're just trying to inflate coverage numbers to pass the coverage test - we will fix it to include integration 
tests, please focus on making sure the code will work in real world not coverage number.